processing/src/test/java/org/apache/druid/frame/field/StringFieldWriterTest.java

```diff
@@ -106,6 +106,12 @@ public void testMultiValueStringContainingNulls()
     doTest(Arrays.asList("foo", NullHandling.emptyToNullIfNeeded(""), "bar", null));
   }
 
+  @Test
```

Comment:
This name is misleading because it tests "nulls" instead of "null bytes", and perhaps not even that. The string should be like "foo\u0000" in order for it to have the null byte. Does this test fail without the changes to the `StringFieldWriter?`

Reply:
Just pushed up a new patch.